pkg/api/types.go

```diff
@@ -47,6 +47,7 @@ type Properties struct {
 	AgentPoolProfiles       []*AgentPoolProfile      `json:"agentPoolProfiles,omitempty"`
 	LinuxProfile            *LinuxProfile            `json:"linuxProfile,omitempty"`
 	WindowsProfile          *WindowsProfile          `json:"windowsProfile,omitempty"`
+	ExtensionsProfile       []ExtensionProfile       `json:"extensionsProfile"`
```

Comment:
I have a slight preference for making `ExtensionsProfile` a `[]*ExtensionProfile` to match the type definition of `AgentPoolProfiles`. And also for calling it `ExtensionProfiles` to match the plural pattern of `AgentPoolProfiles`.

This would make usage patterns look like this:

```
api.ExtensionProfiles = []*ExtensionProfile{}
    for _, p := range vlabs.ExtensionProfiles {
        apiExtensionProfile := &ExtensionProfile{}
        convertVLabsExtensionProfile(p, apiExtensionProfile)
        api.ExtensionProfiles = append(api.ExtensionProfiles, apiExtensionProfile)
    }
```

It's a tiny bit cleaner (we exchange two `&` for a single `*`), but more importantly, it matches the existing pointer patterns and makes the code overall a bit easier to read.

Reply:
sounds good to be consistent